### ISSUE TITLE
Custom HTTP client

### ISSUE THREAD

**Issue by menih** [yavijava/yavijava#276]
YAVIJAVA claim to allow alternative HTTP client, how is that achieved?

**Comment by michaelrice** [comment]
There is an example in the tree, check out the [ApacheHttpClient.java](https://github.com/yavijava/yavijava/blob/gradle/src/main/java/com/vmware/vim25/ws/ApacheHttpClient.java) And the [WSClient.java](https://github.com/yavijava/yavijava/blob/gradle/src/main/java/com/vmware/vim25/ws/WSClient.java)

**Comment by michaelrice** [comment]
Im guessing you may need an example of how to use it once implemented as well. For that see the following test: [ApacheHttpClientTest.java](https://github.com/yavijava/yavijava/blob/gradle/src/intTest/java/com/vmware/vim25/ws/ApacheHttpClientTest.java)

**Comment by menih** [comment]
But how to I swap the client and make the library use one vise the other?

**Comment by michaelrice** [comment]
Did you look at the test I linked to? It provides usage..

**Comment by menih** [comment]
Yes. I have. I was hoping to use abstraction API while underlying HTTP
client is swapped, and not HTTP client directly. I think I figured it
out... All that is needed is this:

        ClientCreator.clientClass = ApacheHttpClient.class;
